server/game/actions/PlayUpgradeAction.ts

```diff
@@ -29,12 +29,18 @@ export class PlayUpgradeAction extends PlayCardAction {
             onPlayCardSource: context.onPlayCardSource,
             playType: context.playType
         });
-        context.game.openEventWindow([
-            context.game.actions
-                .attachUpgrade({ upgrade: context.source, takeControl: context.source.controller !== context.player })
-                .generateEvent(context.target, context),
-            cardPlayedEvent
-        ], this.resolveTriggersAfter);
+        const events = [context.game.actions
+            .attachUpgrade({ upgrade: context.source, takeControl: context.source.controller !== context.player })
+            .generateEvent(context.target, context),
+        cardPlayedEvent];
+
+        if (this.playType === PlayType.Smuggle) {
```

Comment:
Will discuss in chat, but this code should be moved into some kind of base class instead of being repeated in three places